src/inmanta/agent/cache.py

```diff
@@ -40,12 +40,21 @@ def __init__(self, timeout: int = 24 * 3600, version: int = 0) -> None:
 
 
 class CacheItem:
-    def __init__(self, key: str, scope: Scope, value: Any, call_on_delete: Optional[Callable[[Any], None]]) -> None:
+    def __init__(
+        self,
+        key: str,
+        scope: Scope,
+        value: Any,
+        call_on_delete: Optional[Callable[[Any], None]],
+        freshness_period: float,
+    ) -> None:
         self.key = key
         self.scope = scope
         self.value = value
         self.time: float = time.time() + scope.timeout
         self.call_on_delete = call_on_delete
+        self.last_used_at = datetime.datetime.now().astimezone()
```

Comment:
Changed this to:
- the cache item will mark itself as stale after the given `freshness_period`
- this will put this key in a list of keys to cleanup

During cleanup, under exclusive work_in_progress lock:
- go through this list and remove this item from the cache (if they're still in it)

During work (deploy/dry_run/get_facts), under exclusive work_in_progress lock:
- if we hit a stale item, synchronously remove it from the cache before recreating it